internal/appsec/listener/waf/waf.go

```diff
@@ -108,10 +115,9 @@ func (waf *Feature) onFinish(op *waf.ContextOperation, _ waf.ContextRes) {
 	ctx.Close()
 
 	AddWAFMonitoringTags(op, waf.handle.Diagnostics().Version, ctx.Stats().Metrics())
-	if err := SetEventSpanTags(op, op.Events()); err != nil {
-		log.Debug("appsec: failed to set event span tags: %v", err)
+	if wafEvents := op.Events(); len(wafEvents) > 0 {
+		op.SetSerializableTag("_dd.appsec.json", map[string][]any{"triggers": op.Events()})
```

Comment:
serialized in msgpack :-O over meta_struct?

Reply:
No no no, it's just a wrapper for json tags in appsec. It was already used for derivatives, I am simply reusing it